jsk_recognition_utils/src/pcl/ear_clipping_patched.cpp

```diff
@@ -75,77 +75,56 @@ pcl::EarClippingPatched::triangulate (const Vertices& vertices, PolygonMesh& out
     return;
   }
 
-  std::vector<uint32_t> remaining_vertices (n_vertices);
-  if (area (vertices.vertices) > 0) // clockwise?
-    remaining_vertices = vertices.vertices;
-  else
-    for (int v = 0; v < n_vertices; v++)
+  std::vector<uint32_t> remaining_vertices = vertices.vertices;
+  size_t count = triangulate(remaining_vertices, output);
+
+  // if the input vertices order is anti-clockwise, it always left a
+  // convex polygon and start infinite loops, which means will left more
+  // than 3 points.
+  if(remaining_vertices.size() < 3)return;
+
+  output.polygons.erase(output.polygons.end(), output.polygons.end() + count);
+  remaining_vertices.resize(n_vertices);
+  for (size_t v = 0; v < n_vertices; v++)
       remaining_vertices[v] = vertices.vertices[n_vertices - 1 - v];
+  triangulate(remaining_vertices, output);
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+size_t
+pcl::EarClippingPatched::triangulate (std::vector<uint32_t>& vertices, PolygonMesh& output)
```

Comment:
I think it is better to name different function name from `triangulate` because it may cause misunderstanding.
how about `triangulate_step` ?

Reply:
Renamed: `triangulate` -> `triangulateClockwiseVertices`.